pkg/lib/sms/cmcom/send.go

```diff
@@ -0,0 +1,230 @@
+package cmcom
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/authgear/authgear-sms-gateway/pkg/lib/api"
+	"github.com/authgear/authgear-sms-gateway/pkg/lib/sensitive"
+	"github.com/authgear/authgear-sms-gateway/pkg/lib/sms/smsclient"
+)
+
+const CM_API_ENDPOINT = "https://gw.cmtelecom.com/v1.0/message"
+
+type MessageBody struct {
+	Type    string `json:"type"`
+	Content string `json:"content"`
+}
+
+type To struct {
+	Number string `json:"number"`
+}
+
+type Msg struct {
+	From      string      `json:"from"`
+	To        []To        `json:"to"`
+	Body      MessageBody `json:"body"`
+	Reference string      `json:"reference,omitempty"`
+}
+
+type Messages struct {
+	Msg []Msg `json:"msg"`
+}
+
+type SendRequest struct {
```

Comment:
Can you add a comment here to point to the doc you referenced in this implementation?